Source/ASVideoNode.mm

```diff
@@ -467,15 +467,24 @@ - (void)didExitVisibleState
 {
   [super didExitVisibleState];
   
-  ASDN::MutexLocker l(__instanceLock__);
-  
-  if (_shouldBePlaying) {
+  __instanceLock__.lock();
+  // We have to cache the shouldBePlaying state here as pause will set it to NO
+  BOOL shouldBePlaying = _shouldBePlaying;
+
+  if (shouldBePlaying) {
+    __instanceLock__.unlock();
```

Comment:
This seems pretty complicated - could we just leave it locked here? Otherwise we would lose transactionality and then the subtle need to cache the value of shouldBePlaying in a local variable comes into play.

I think that we will probably never get to a stage where __instanceLock__ is a non-recursive lock, so it might not be beneficial to avoid calling pause while locked.